src/Schedule.php

```diff
@@ -53,25 +53,27 @@ public function materialize(\DateTimeImmutable $from, \DateInterval $period): ar
     }
 
     /**
-     * @return array
+     * @param int $id
      */
-    public function getData(): array
+    public function persists(int $id): void
     {
-        $rules = [];
-        foreach ($this->rules as $rule) {
-            $object = new \stdClass();
-            $object->type = get_class($rule);
-            $object->data = json_encode($rule);
-            $rules[] = $object;
-        }
-        return $rules;
+        $this->id = $id;
     }
 
     /**
-     * @param int $id
+     * @return \stdClass
      */
-    public function persists(int $id): void
+    public function jsonSerialize(): \stdClass
     {
-        $this->id = $id;
+        $data = new \stdClass();
+        $data->id = $this->id;
+        $data->rules = [];
+        foreach ($this->rules as $rule) {
+            $object = new \stdClass();
+            $object->type = get_class($rule);
+            $object->data = json_encode($rule);
```

Comment:
I've waited for this :)

The mistake is calling `jsonSerialize` method where you want to serialize a Schedule as JSON. Please read the documentation: https://www.php.net/manual/en/class.jsonserializable.php

* Right way: $obj = json_decode(json_encode(Schedule))
* Wrong way: $obj = $schedule->jsonSerialize()

When you call `jsonSerialize()` you won't get nested objects serialized.